### ISSUE TITLE
Gradle jar/war configuration is a little unusual

### ISSUE THREAD

**Issue by wilkinsona** [spring-io/initializr#381]
When generating a Gradle project, the `build.gradle` contains the following configuration of the `jar` task:

```groovy
jar {
        baseName = 'demo'
        version = '0.0.1-SNAPSHOT'
}
```

I think this is rather unusual for a couple of reasons:

1. The version is typically "inherited" from the project rather than being specific to the jar task. Configuring the version on the project would be more idiomatic
2. The name is typically "inherited" from the project rather than being specific to the jar task. The project's name can't be set in `build.gradle`, rather it's controlled by the name of the directory that contains the `build.gradle`. The downloaded zip already contains a directory with the right name so this configuration could be removed entirely.

In short, I think the section above could be replaced with:

```groovy
version = '0.0.1-SNAPSHOT'
```

**Comment by dsyer** [comment]
The downloaded zip doesn't always contain a directory. Should we keep the `baseName`?

**Comment by wilkinsona** [comment]
I wondered about that, but it'd lead to the odd situation where the project has one name and the jar has another. On the other hand, you might have the odd situation where you specified one name when generating the project and got another when you actually built it. I think the second it the least bad.
